src/core/components/hv-doc/index.tsx

```diff
@@ -39,12 +40,36 @@ export default class HvDocState extends PureComponent<
       styles: null,
       url: null,
     };
+    this.backBehaviorElementCache = [];
   }
 
   render(): React.ReactNode {
     return (
-      <Contexts.DocStateContext.Provider
+      <Contexts.DocContext.Provider
         value={{
+          backBehaviorElements: {
```

Comment:
Following my other comment, the instance of `BackBehaviorElements` could be created in the constructor of this component, stored on an instance property and referenced here - that way we don't recreate the object with inner functions on every render cycle.

Reply:
Done